allure-support/src/main/kotlin/com/kaspersky/components/alluresupport/interceptors/testrun/VideoRecordingTestInterceptor.kt

```diff
@@ -2,18 +2,39 @@ package com.kaspersky.components.alluresupport.interceptors.testrun
 
 import com.kaspersky.components.alluresupport.files.attachVideoToAllureReport
 import com.kaspersky.kaspresso.device.video.Videos
+import com.kaspersky.kaspresso.files.dirs.AllureDirsProvider
 import com.kaspersky.kaspresso.interceptors.watcher.testcase.TestRunWatcherInterceptor
 import com.kaspersky.kaspresso.testcases.models.info.TestInfo
+import io.qameta.allure.kotlin.Allure
+import io.qameta.allure.kotlin.model.Status
 
+/**
+ * Due to screen recorder bug we have to perform a workaround which requires VideoRecordingTestInterceptor
+ * and MoveReportsInterceptor to be the last two interceptors in allure reports. So if you use VideoRecordingTestInterceptor
+ * be sure to use MoveReportsInterceptor too. Otherwise your report will be under /data/data/your.package.name/files/allure-results
+ * and videos under /sdcard
+ */
 class VideoRecordingTestInterceptor(
-    private val videos: Videos
+    private val videos: Videos,
+    private val allureDirsProvider: AllureDirsProvider,
+    private val stateHolder: TestRunStateHolder
 ) : TestRunWatcherInterceptor {
 
     override fun onTestStarted(testInfo: TestInfo) {
         videos.record("Video_${testInfo.testName}")
     }
 
     override fun onTestFinished(testInfo: TestInfo, success: Boolean) {
-        videos.saveAndApply { attachVideoToAllureReport() }
+        videos.saveAndApply {
+            val stubFile = allureDirsProvider.provideReportVideoAttachmentStub(this)
+            stubFile.attachVideoToAllureReport()
+            val uuid = Allure.lifecycle.getCurrentTestCase() ?: ""
```

Comment:
okey, we can get this UUID after test, but in case below we store UUID in a field before test, why?

Reply:
I've encountered situation when Allure.lifecycle.getCurrentTestCase() returned null or wrong UUID if there were multiple tests when I call it in onTestFinished

Reply:
hmmm, i think we need more info about this situation, may be it is allure bug?